client/lib/feed-stream-store/index.js

```diff
@@ -45,6 +45,11 @@ function mixedKeyMaker( post ) {
 	return siteKeyMaker( post );
 }
 
+const recommendedKeyMaker = post => ( {
+	...siteKeyMaker( post ),
+	isRecommendation: true }
```

Comment:
i think this is something different.

postKeys should represent either posts or gaps, not the cards we have in the stream. The stream should render a series of item descriptors, where an item descriptor could be a plain postKey, or it could be one of these `is*` things. But I really don't want to call those post keys and clutter the API to the feed-post-store or the stream-store.